spec/scenarios/invoices/invoice_numbering_spec.rb

```diff
@@ -309,4 +315,435 @@
+            external_id: customer_second.external_id,
+            plan_code: monthly_plan.code,
+            billing_time: 'anniversary',
+            subscription_at: subscription_at.iso8601,
+          },
+        )
+        create_subscription(
+          {
+            external_customer_id: customer_third.external_id,
+            external_id: customer_third.external_id,
+            plan_code: monthly_plan.code,
+            billing_time: 'anniversary',
+            subscription_at: subscription_at.iso8601,
+          },
+        )
+
+        invoices = organization.invoices.order(created_at: :desc).limit(3)
+        sequential_ids = invoices.pluck(:sequential_id)
+        organization_sequential_ids = invoices.pluck(:organization_sequential_id)
+        numbers = invoices.pluck(:number)
+
+        expect(sequential_ids).to match_array([1, nil, 1])
+        expect(organization_sequential_ids).to match_array([0, 0, 0])
+        expect(numbers).to match_array(%w[ORG-1-001-001 ORG-1-DRAFT ORG-1-003-001])
+      end
+
+      # NOTE: Jul 20th: New subscription for the first customer
+      time = subscription_at + 1.day
+      travel_to(time) do
+        create_subscription(
+          {
+            external_customer_id: customer_first.external_id,
+            external_id: 'new_external_id',
+            plan_code: yearly_plan.code,
+            billing_time: 'anniversary',
+            subscription_at: time.iso8601,
+          },
+        )
+
+        invoices = organization.reload.invoices.order(created_at: :desc)
+
+        expect(invoices.first.sequential_id).to eq(2)
+        expect(invoices.first.organization_sequential_id).to eq(0)
+        expect(invoices.pluck(:number))
+          .to match_array(
+            %w[
+              ORG-1-001-001
+              ORG-1-DRAFT
+              ORG-1-003-001
+              ORG-1-001-002
+            ],
+          )
+      end
+
+      # NOTE: Jul 21st: New subscription for the second customer
+      time = subscription_at + 2.days
+      travel_to(time) do
+        create_subscription(
+          {
+            external_customer_id: customer_second.external_id,
+            external_id: 'new_external_id_2',
+            plan_code: yearly_plan.code,
+            billing_time: 'anniversary',
+            subscription_at: time.iso8601,
+          },
+        )
+
+        invoices = organization.reload.invoices.order(created_at: :desc)
+
+        expect(invoices.first.sequential_id).to eq(nil)
+        expect(invoices.first.organization_sequential_id).to eq(0)
+        expect(invoices.pluck(:number))
+          .to match_array(
+            %w[
+              ORG-1-001-001
+              ORG-1-DRAFT
+              ORG-1-003-001
+              ORG-1-001-002
+              ORG-1-DRAFT
+            ],
+          )
+      end
+
+      travel_to(time + 1.hour) do
+        draft_invoice_1 = customer_second.reload.invoices.draft.order(created_at: :asc).first
```

Comment:
Naming/VariableNumber: Use normalcase for variable numbers. (https://rubystyle.guide#snake-case-symbols-methods-vars-with-numbers)